### PR TITLE
Refactor uninstall controller for structured log

### PR DESCRIPTION
**Pull request by c3y1huang**
Refactor setting controller to address structured logging implementation.

https://github.com/longhorn/longhorn/issues/1733
https://github.com/longhorn/longhorn-manager/pull/661

### CODE REVIEW

File: controller/uninstall_controller.go [hunk 1]
```diff
@@ -470,17 +485,19 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
 		err = errors.Wrapf(err, "Failed to delete instance managers")
 	}()
 	for _, im := range instanceManagers {
-		logFields := logrus.Fields{
-			"type": "instanceManager",
-			"name": im.Name,
-		}
+		log := c.logger.WithFields(
```
Comment (joshimoo) [inline controller/uninstall_controller.go:488]: Add a getLoggerForInstanceManager function to the instance-manager-controller, same idea as above for the engine.
The idea is to have a getLoggerX for each crd type.

File: controller/uninstall_controller.go [hunk 2]
```diff
@@ -491,50 +508,62 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
...
 			}
-			logrus.WithFields(logFields).Infof("Removed finalizer")
+			log.Info("Removed finalizer")
 		}
 	}
 	return
 }
 
 func (c *UninstallController) deleteManager() (bool, error) {
+	log := c.logger.WithFields(
```
Comment (joshimoo) [inline controller/uninstall_controller.go:525]: For the DaemonSet/ReplicaSet you can add the getLoggerForX functions to this controller for now.
the idea is that the fieldName is `daemonSet`, with the value of the daemonsSet name.

File: controller/uninstall_controller.go [hunk 3]
```diff
@@ -551,54 +580,72 @@ func (c *UninstallController) deleteDriver() (bool, error) {
 	}
 	wait := false
 	for _, name := range deploymentsToClean {
+		log := c.logger.WithFields(
```
Comment (joshimoo) [inline controller/uninstall_controller.go:583]: See above for deamonSet

**Review by joshimoo — CHANGES_REQUESTED** [review]
Thanks for the pr, I added some suggestions.

**Comment by c3y1huang** [comment]
> Thanks for the pr, I added some suggestions.

Thanks for the review, I've addressed all in https://github.com/longhorn/longhorn-manager/pull/706/commits/0cf8db6d12d8453bc601cdd320a27f1f4db6abaf.

**Review by joshimoo — APPROVED** [review]
LGTM, for the daemonset/deployment case you could have used references to the actual deployment/daemonset but since we currently only use this in the uninstall controller we can refactor that part later :)

**Comment by joshimoo** [comment]
@c3y1huang Thanks for the PR and the addition of the new logger functions :)

**Pull request merged**